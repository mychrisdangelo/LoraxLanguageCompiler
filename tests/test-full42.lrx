/* 
 * Author: Chris D'Angelo
 * Stress testing plus operator with [] edge cases
 */

int main()
{
    tree <int> t(2);
    tree <int> s(2);
    tree <int> m(1);

    t = 6[] + 4[2, 3];
    print(t, "\n");
    m = 4[] + 6[];
    print(t, "\n");
    t = 4[2, 3] + 6[];
    print(t, "\n");

    t = 4[2[5,6], 3] + 6[7, 8];
    print(t, "\n");

    s = 5[10, 9];
    t = t + s + s + t ;
    print(t, "\n");
}