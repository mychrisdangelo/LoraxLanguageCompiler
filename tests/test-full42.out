6[4[2[null,null],3[null,null]],null]
6[4[2[null,null],3[null,null]],null]
4[2[6[null,null],null],3[null,null]]
4[2[5[null,null],6[null,null]],3[6[7[null,null],8[null,null]],null]]
4[2[5[5[10[null,null],9[null,null]],4[2[5[null,null],6[null,null]],3[6[7[null,null],8[null,null]],null]]],6[null,null]],3[6[7[null,null],8[null,null]],5[10[null,null],9[null,null]]]]
